waspc/src/Wasp/Lib.hs

```diff
@@ -96,6 +68,45 @@ warnIfDotEnvPresent waspDir = do
     Nothing -> return []
     Just _ -> return ["Wasp .env files should be named .env.server or .env.client, depending on their use."]
 
+findWaspFilePath :: Path' Abs (Dir WaspProjectDir) -> IO (Either (NonEmpty CompileError) (Path' Abs File'))
+findWaspFilePath waspDir = do
+  maybeWaspFilePath <- findWaspFile waspDir
+  case maybeWaspFilePath of
+    Nothing -> return $ Left $ fromList ["Couldn't find a single *.wasp file."]
+    Just waspFilePath -> return $ return waspFilePath
+
+analyzeWaspFileContent :: Path' Abs File' -> IO (Either (NonEmpty CompileError) [Decl])
+analyzeWaspFileContent waspFilePath = do
+  waspFileContent <- readFile (SP.fromAbsFile waspFilePath)
+  return $ case Analyzer.analyze waspFileContent of
+    Right decls -> Right decls
+    Left analyzeError -> Left $ fromList
+          [ showCompilerErrorForTerminal
+              (waspFilePath, waspFileContent)
+              (getErrorMessageAndCtx analyzeError)
+          ]
+
+createAppSpec :: Path' Abs (Dir WaspProjectDir) -> CompileOptions -> [Decl] -> IO AS.AppSpec
+createAppSpec waspDir options decls = do
+  externalServerCodeFiles <-
+    ExternalCode.readFiles (CompileOptions.externalServerCodeDirPath options)
+  externalClientCodeFiles <-
+    ExternalCode.readFiles (CompileOptions.externalClientCodeDirPath options)
+  maybeDotEnvServerFile <- findDotEnvServer waspDir
+  maybeDotEnvClientFile <- findDotEnvClient waspDir
```

Comment:
@shayneczyzewski Do you think it makes sense to move `.env` files into `server` and `client` directories? It might be a bit cleaner.

Reply:
Sounds good to me also!

Reply:
We'll do this in a different issue, I'll make one before merging the PR.

Reply:
Did this issue get created BTW?

Reply:
Not yet, I usually go through all unresolved threads and create issues while waiting for the pre-merge CLI to do its thing.

Reply:
Done: #810

Reply:
Done: #811.